odnoklassniki-android-sdk/src/main/java/ru/ok/android/sdk/Odnoklassniki.java

```diff
@@ -404,20 +405,19 @@ public void run() {
      * @param args     widget arguments as specified in documentation
      * @param postingListener - listener which will be called after method call
      */
-    public void performPosting(String attachment, boolean userTextEnabled,
+    public void performPosting(Activity activity, String attachment, boolean userTextEnabled,
                                @Nullable HashMap<String, String> args,
                                OkListener postingListener) {
         this.mOkListener = postingListener;
 
-        Intent intent = new Intent(mContext, OkPostingActivity.class);
+        Intent intent = new Intent(activity, OkPostingActivity.class);
         intent.putExtra(Shared.PARAM_APP_ID, mAppId);
         intent.putExtra(Shared.PARAM_ATTACHMENT, attachment);
         intent.putExtra(Shared.PARAM_ACCESS_TOKEN, mAccessToken);
         intent.putExtra(Shared.PARAM_WIDGET_ARGS, args);
         intent.putExtra(Shared.PARAM_SESSION_SECRET_KEY, mSessionSecretKey);
         intent.putExtra(Shared.PARAM_USER_TEXT_ENABLE, userTextEnabled);
-        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-        mContext.startActivity(intent);
+        activity.startActivityForResult(intent, OkAuthActivity.OK_AUTH_REQUEST_CODE);
```

Comment:
In the means of using the same listener for different actions - no, it did not. However, using startActivityForResult can provide better API for the client code, so why keep these old restrictions?